src/en/examples/mobile/compass.js

```diff
@@ -86,22 +86,13 @@ const style = new Style({
 layer.setStyle(style);
 //! [style]
 //! [kompas]
-if (
-  window.DeviceOrientationEvent &&
-  typeof DeviceOrientationEvent.requestPermission === 'function'
-) {
-  locate.addEventListener('click', function () {
-    DeviceOrientationEvent.requestPermission()
-      .then(function () {
-        const compass = new Kompas();
-        compass.watch();
-        compass.on('heading', function (heading) {
-          style.getImage().setRotation((Math.PI / 180) * heading);
-        });
-      })
-      .catch(function (error) {
-        alert(`ERROR: ${error.message}`);
-      });
-  });
+if ('ondeviceorientationabsolute' in window) {
+  kompas()
+    .watch()
+    .on('heading', function (heading) {
+      style.getImage().setRotation((Math.PI / 180) * heading);
+    });
+} else {
+  alert('No absolute orientation provided by device');
```

Comment:
As far as I remember, to make this work in both Chrome and Safari, you'd have to do something like
```js
function startCompass() {
  kompas()
    .watch()
    .on('heading', function (heading) {
      style.getImage().setRotation((Math.PI / 180) * heading);
    });
}

if (
  window.DeviceOrientationEvent &&
  typeof DeviceOrientationEvent.requestPermission === 'function'
) {
  locate.addEventListener('click', function () {
    DeviceOrientationEvent.requestPermission()
      .then(startCompass)
      .catch(function (error) {
        alert(`ERROR: ${error.message}`);
      });
  });
} else if ('ondeviceorientationabsolute' in window) {
  startCompass();
} else {
  alert('No device orientation provided by device');
}
```

Reply:
Indeed, tried it in an emulator and I don't get the warning in Safari; since I don't have a real device I can't really test that it works though